components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -2106,6 +2107,17 @@ private IdentityProvider getIDP(Connection dbConnection, String idPName, int idp
                 federatedIdp.setFederatedAuthenticatorConfigs(getFederatedAuthenticatorConfigs(
                         dbConnection, idPName, federatedIdp, tenantId));
 
+                // Retrieve encrypted secrets from DB, decrypt and set to the federated authenticator configs.
+                if (IdpMgtServiceComponentHolder.getInstance().getIdPSecretsProcessorService() == null) {
+                    throw new IdentityProviderManagementException(
+                            "Error while retrieving secrets of identity provider: " + idPName + " in tenant: " +
+                                    tenantDomain + ". IdPSecretsProcessorService is not available.");
+                }
+                if (federatedIdp.getFederatedAuthenticatorConfigs().length > 0) {
```

Comment:
`federatedIdp.getFederatedAuthenticatorConfigs()`, Is there any chance to get the NPE?

Reply:
This will return an empty array if there is no fedAuthConfigs